app/src/androidTest/java/ch/epfl/sdp/drone3d/service/storage/dao/FirebaseMappingMissionDaoTest.kt

```diff
@@ -13,40 +13,56 @@ import ch.epfl.sdp.drone3d.service.impl.storage.dao.FirebaseMappingMissionDao
 import com.google.firebase.database.ChildEventListener
 import com.google.firebase.database.DataSnapshot
 import com.google.firebase.database.DatabaseError
-import com.google.firebase.database.ktx.database
+import com.google.firebase.database.FirebaseDatabase
 import com.google.firebase.database.ktx.getValue
-import com.google.firebase.ktx.Firebase
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
 import org.hamcrest.CoreMatchers.equalTo
 import org.hamcrest.MatcherAssert.assertThat
-import org.hamcrest.Matchers.*
+import org.hamcrest.Matchers.nullValue
+import org.junit.After
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import java.util.concurrent.CountDownLatch
 import java.util.concurrent.TimeUnit
+import javax.inject.Inject
 
+@HiltAndroidTest
 class FirebaseMappingMissionDaoTest {
 
-    @get:Rule
-    val instantTaskExecutorRule = InstantTaskExecutorRule()
-
-    private val database =
-        Firebase.database("https://drone3d-6819a-default-rtdb.europe-west1.firebasedatabase.app/")
-
-    private val db = FirebaseMappingMissionDao(database)
-
-    private val timeout = 5L
-
     companion object {
         private const val OWNERID: String = "Jean-Jean"
         private val MAPPING_MISSION_1: MappingMission = MappingMission()
         private val MAPPING_MISSION_2: MappingMission = MappingMission()
     }
 
+    @get:Rule
+    val rule = HiltAndroidRule(this)
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @Inject lateinit var database: FirebaseDatabase
+
+    private lateinit var db: FirebaseMappingMissionDao
+    private val timeout = 5L
```

Comment:
Should be in companion object rather than here

Reply:
Good point, I moved it